File: samples/matlab/s024_plot_geometry.m
%% s024_plot_geometry.m
% brief             example of usage for astra_plot_geom command
% - last update     16.11.2018
% -----------------------------------------------------------------------
% This file is part of the ASTRA Toolbox
% 
% Copyright: 2010-2018, imec Vision Lab, University of Antwerp
%            2014-2018, CWI, Amsterdam
% License: Open Source under GPLv3
% Contact: astra@astra-toolbox.com
% Website: http://www.astra-toolbox.com/
% -----------------------------------------------------------------------
%%
close all;

if exist('astra_create_example_cone') ~= 2
    error('Please add astra/algorithms/plot_geom to your path to use this function')
end


% proj_geom = astra_create_example_cone('vec');
% proj_geom = astra_create_example_cone('normal');
proj_geom = astra_create_example_cone('helix');
% proj_geom = astra_create_example_parallel3d('vec');
% proj_geom = astra_create_example_fanflat('vec');
% proj_geom = astra_create_example_fanflat();
% proj_geom = astra_create_example_parallel3d();
% proj_geom = astra_create_example_cone('deform_vec');

astra_plot_geom(proj_geom);
hold on;

vol_magn = 20;
phantom_size = 5;
phantom_px = 1500;
vx_size = phantom_size / phantom_px; % voxel size
vol_geom = astra_create_vol_geom(phantom_px, phantom_px, phantom_px);
line_width = 1; % line width for phantom
astra_plot_geom(vol_geom, vx_size, 'Magnification', vol_magn,...
    'LineWidth', line_width, 'Color', 'r');

% this magnification is empirically chosen to fit the stl file
cad_magn = 900;
astra_plot_geom('bunny.stl', cad_magn);

hold off;
axis equal;

File: samples/python/s024_plot_geometry.py
# -----------------------------------------------------------------------
#   brief             example of usage for astra_plot_geom command
#   - last update     08.05.2019
# -----------------------------------------------------------------------
# Copyright: 2010-2019, imec Vision Lab, University of Antwerp
#            2013-2019, CWI, Amsterdam
#
# Contact: astra@astra-toolbox.com
# Website: http://www.astra-toolbox.com/
#
# This file is part of the ASTRA Toolbox.
#
# The ASTRA Toolbox is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# The ASTRA Toolbox is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with the ASTRA Toolbox. If not, see <http://www.gnu.org/licenses/>.
#
# -----------------------------------------------------------------------
import astra
import numpy as np
import matplotlib.pyplot as plt

# proj_geom = astra.create_example('cone', 'normal')
# proj_geom = astra.create_example('cone', 'vec')
proj_geom = astra.create_example('cone', 'helix')
# proj_geom = astra.create_example('cone', 'deform_vec')
# proj_geom = astra.create_example('parallel3d', 'vec')
# proj_geom = astra.create_example('parallel3d')
# proj_geom = astra.create_example('fanflat', 'vec')
# proj_geom = astra.create_example('fanflat')
ax =  astra.plot_geom(proj_geom)

vol_magn = 20
phantom_size = 5.0
phantom_px = 1500.0
vx_size = phantom_size / phantom_px  # voxel size
vol_geom = astra.create_vol_geom(phantom_px, phantom_px, phantom_px)
astra.plot_geom(vol_geom, ax, vx_size=vx_size, Magnification=vol_magn, Color='r')

# the cad model plotting is not supported in python

plt.show()
